https-upgrades/tests.json

```diff
@@ -0,0 +1,158 @@
+{
+    "navigations": {
+        "name": "Navigations",
+        "desc": "Navigations should be upgraded to https",
+        "tests": [
+            {
+                "name": "bloom filter match - should be upgraded",
+                "siteURL": "http://stackoverflow.com/path/file.html?search=query&another=one#fragment",
+                "requestURL": "http://stackoverflow.com/path/file.html?search=query&another=one#fragment",
+                "requestType": "main_frame",
+                "expectURL": "https://stackoverflow.com/path/file.html?search=query&another=one#fragment",
+                "exceptPlatforms": []
+            },
+            {
+                "name": "unknown subdomain - should not be upgraded",
+                "siteURL": "http://fake.stackoverflow.com/path/file.html?search=query&another=one#fragment",
+                "requestURL": "http://fake.stackoverflow.com/path/file.html?search=query&another=one#fragment",
+                "requestType": "main_frame",
+                "expectURL": "http://fake.stackoverflow.com/path/file.html?search=query&another=one#fragment",
+                "exceptPlatforms": []
+            },
+            {
+                "name": "bloom filter allowlist match - should not be upgraded",
+                "siteURL": "http://hobbyncoffee.com/path/file.html?search=query&another=one#fragment",
+                "requestURL": "http://hobbyncoffee.com/path/file.html?search=query&another=one#fragment",
+                "requestType": "main_frame",
+                "expectURL": "http://hobbyncoffee.com/path/file.html?search=query&another=one#fragment",
+                "exceptPlatforms": []
+            },
+            {
+                "name": "already https - should not be modified",
+                "siteURL": "https://youtube.com/path/file.html?search=query&another=one#fragment",
+                "requestURL": "https://youtube.com/path/file.html?search=query&another=one#fragment",
+                "requestType": "main_frame",
+                "expectURL": "https://youtube.com/path/file.html?search=query&another=one#fragment",
+                "exceptPlatforms": []
+            },
+            {
+                "name": "negative bloom filter match - should not be upgraded",
```

Comment:
Some of these are hard to verify from the test data since I don't think I can decode the bloom filter datasets?  I'll assume this will sort itself out in the test implementation, but it might be better to build the bloom filter datasets on the fly from only a short list of domains we intend to test?

Reply:
generating it on the fly in this repo might be a bit weird, but I agree that a smaller bloom filters with just couple test domains would be better. I'll try to find out what it takes to create such a bloom filter by hand.